FreydCategoriesForCAP/gap/FreydCategory.gi

```diff
@@ -19,64 +19,47 @@ InstallMethod( FreydCategory,
   function( underlying_category )
     local freyd_category, to_be_finalized;
     
-    if not HasIsAdditiveCategory( underlying_category ) then
-        
-        Error( "The given category should be additive" );
-        
-    fi;
-    
-    ## for IsCongruentForMorphisms to be correct
-    if not CanCompute( underlying_category, "Lift" ) then
-        
-        Error( "The given category should be able to compute Lift" );
-        
-    fi;
-    
-    if not CanCompute( underlying_category, "SubtractionForMorphisms" ) then
-        
-        Error( "The given category should be able to compute SubtractionForMorphisms" );
-        
-    fi;
-    
-    if not CanCompute( underlying_category, "PreCompose" ) then
-        
-        Error( "The given category should be able to compute PreCompose" );
-        
-    fi;
-    
-    freyd_category := CreateCapCategory( Concatenation( "Freyd( ", Name( underlying_category ), " )" ) );
-    
-    SetFilterObj( freyd_category, IsFreydCategory );
-    
-    SetIsAdditiveCategory( freyd_category, true );
-    
-    SetUnderlyingCategory( freyd_category, underlying_category );
-    
-    ## Freyd's theorem
-    if ForAll( [ "WeakKernelEmbedding", "WeakKernelLift" ], f -> CanCompute( underlying_category, f ) ) then
-        
-        SetIsAbelianCategory( freyd_category, true );
-        
-    fi;
-    
-    AddObjectRepresentation( freyd_category, IsFreydCategoryObject );
-    
-    AddMorphismRepresentation( freyd_category, IsFreydCategoryMorphism );
+    if IsValidInputForFreydCategory( underlying_category ) then
```

Comment:
Could you change the logic to:
```
if not IsValidInputForFreydCategory( underlying_category ) then
    return false;
fi;
```
and then adjust the indenting accordingly?
In this way, the code is easier for me to read.

Reply:
Yes, the commit below provides this change.